### PR TITLE
Remove tar support

### PR DESCRIPTION
**Pull request by Morganamilo**
tar is slower and harder to diff. I doubt any one still uses it intentially.

There's a chance some people have not cleared their cache in two years
and still have tar based packages around. But forcing them to clear
their cache is not a big deal at all.

### CODE REVIEW

**Comment by Morganamilo** [comment]
I ended up removing `-f` but looks like we still need it for ABS packages. Even though we use git now we still delete the dir and copy it over.

**Comment by Morganamilo** [comment]
Also reading through this, do we really need to shell out to `cp`?

**Comment by Jguer** [comment]
> Also reading through the ABS code, do we really need to shell out to `cp`?

A inheritance from the old inline tar.gz code, can be removed.
https://stackoverflow.com/questions/51779243/copy-a-folder-in-go

**Comment by Jguer** [comment]
#1102 now implements the changes to bash and zsh completion for this

File: parser.go [hunk 1]
```diff
@@ -528,10 +526,6 @@ func handleConfig(option, value string) bool {
 		config.AnswerUpgrade = value
 	case "noanswerupgrade":
 		config.AnswerUpgrade = ""
-	case "gitclone":
-		config.GitClone = true
-	case "nogitclone":
```
Comment (Jguer) [inline parser.go]: Also missing removal of case "tar" in both attribution and valid param

File: config.go [hunk 2]
```diff
@@ -72,7 +72,6 @@ type Configuration struct {
 	NoConfirm          bool   `json:"-"`
 	Devel              bool   `json:"devel"`
 	CleanAfter         bool   `json:"cleanAfter"`
-	GitClone           bool   `json:"gitclone"`
```
Comment (Jguer) [inline config.go]: Missing removal of `config.Tarbin`

**Comment by Morganamilo** [comment]
Annoyingly when searching for `tar` it also pulls in `nextArg` and `target`. So I couldn't really just do a find all. I was kind of relying on the CI to catch the unused for me.

**Comment by bundi78** [comment]
Hi,

sorry to be late, but for me, that feature was useful. I have to work behind an awkward proxy and --nogitclone was the only working solution to use AUR.

With 9.4.3, I have no (easy) access to AUR.

**Comment by mjruschak** [comment]
Oh boy... Any way you could add this back.  Found this option useful for behind proxys.  Much appreciated if we could get it back.

**Pull request merged**